### PR TITLE
CDAP-4785 protected against null pointer exception in upgrade

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java [hunk 1]
```diff
@@ -98,6 +98,12 @@ public void upgrade() throws Exception {
               for (Map.Entry<Long, byte[]> columnEntry : columnMap.getValue().entrySet()) {
                 Long timeStamp = columnEntry.getKey();
                 byte[] colVal = columnEntry.getValue();
+                // a deleted dataset can still show up here since BufferingTable doesn't actually delete, but
+                // writes a null value. The fact that we need to know that implementation detail here is bad.
+                // If we could use Table here instead of HTable, this would be hidden from us.
+                if (colVal == null || colVal.length == 0) {
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: I think we should do a similar check in other upgrades.

Comment (albertshau) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: do you know where else we do stuff like this?

Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: AbstractQueueUpgrader

Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: @rohitsinha54 has a task which also might involve this.

Comment (albertshau) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: AbstractQueueUpgrader doesn't do anything with the column value though, it just transfers it directly into the Put

Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java]: I see; ok. Then its probably fine. @rohitsinha54 just has to keep in mind to do something similar in this PR when he works on his upgrade task.

**Comment by anwar6953** [comment]
LGTM

**Pull request merged**